### DISCUSSION TITLE
External packages - what are the limitations? plotly example

### DISCUSSION THREAD

**Discussion by sadruddin** [idom-team/idom#341] [Question] [answered]
Hello,

I've been trying to use react-plotly.js from idom, but while the install seems to work initially, the "idom install react-plotly.js" command ends up displaying:

> successfuly installed ['idom-client-react', 'react', 'react-plotly.js', 'react-dom'] but failed to discover ['react-plotly.js'] post-install

Has anyone encountered a similar problem? Also, more generically, is there a current and/or structural limitation in terms of external JS/react modules characteristics support?

Thanks!

**Comment by rmorshea** [comment, marked as answer]
My initial suggestion would probably to try running `idom restore` and then try again? There's an upcoming release (maybe sometime this week) that may improve or fix this issue. Will try to remember to post back here when that's done.

And there are definitely limitations with respect to using dynamically installed JS packages:

1. At the moment only packages with React components are supported
2. You can only pass JSON serializable props to those React components.
